x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/Dissect.java

```diff
@@ -10,17 +10,42 @@
 import org.elasticsearch.dissect.DissectParser;
 import org.elasticsearch.xpack.esql.core.expression.Attribute;
 import org.elasticsearch.xpack.esql.core.expression.Expression;
+import org.elasticsearch.xpack.esql.core.expression.NameId;
+import org.elasticsearch.xpack.esql.core.expression.ReferenceAttribute;
 import org.elasticsearch.xpack.esql.core.tree.NodeInfo;
 import org.elasticsearch.xpack.esql.core.tree.Source;
+import org.elasticsearch.xpack.esql.core.type.DataType;
+import org.elasticsearch.xpack.esql.parser.ParsingException;
 
+import java.util.ArrayList;
 import java.util.List;
 import java.util.Objects;
+import java.util.Set;
 
 public class Dissect extends RegexExtract {
     private final Parser parser;
 
     public record Parser(String pattern, String appendSeparator, DissectParser parser) {
 
+        public List<Attribute> keyAttributes(Source src) {
```

Comment:
Moving this one here is a bit forced. Especially since it seems to act at the parser level (ie the use of `ParsingException).

Reply:
Moving it - just the validation - back into `LogicalPlanBuilder.visitDissectCommand`.